kairon/shared/chat/agent/agent_flow.py

```
@@ -0,0 +1,287 @@
+                    action_endpoint=self.action_endpoint
+                )
+                new_events_for_tracker = await action.run(output_channel=None,
+                                 nlg=self.nlg,
+                                 tracker=self.fake_tracker,
+                                 domain=self.domain)
+                self.process_tracker_events(new_events_for_tracker)
+                self.fake_tracker.events.extend(new_events_for_tracker)
+                self.executed_actions.append(event['name'])
+            except Exception as e:
+                logger.error(f"Error in executing action [{event['name']}]: {e}")
+                raise AppException(f"Error in executing action [{event['name']}]")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)